### PR TITLE
Add Individual Profile page

### PR DESCRIPTION
**Pull request by subinasr**
depends on https://github.com/toggle-corp/kitab-server/pull/17

### CODE REVIEW

File: app/Base/configs/routes.tsx [hunk 1]
```diff
@@ -40,14 +40,14 @@ const home = wrap({
...
+    path: '/profile/',
     title: 'My Profile',
     navbarVisibility: true,
-    component: lazy(() => import('#views/Template')),
+    component: lazy(() => import('#views/Profile')),
     componentProps: {
         name: 'My Profile Page',
     },
-    visibility: 'is-authenticated',
+    visibility: 'is-anything', // TODO: change visibility to 'is-authenticated'
```
Comment (AdityaKhatri) [inline app/Base/configs/routes.tsx]: why add todo?

File: app/views/Profile/IndividualProfile/index.tsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+                        }
+                    }
+                }
+            }
+
+        }
+    }
+`;
+
+type StatusType = 'RECEIVED' | 'PACKED' | 'COMPLETED' | 'CANCELLED';
```
Comment (AdityaKhatri) [inline app/views/Profile/IndividualProfile/index.tsx]: Get this from generated if possible

File: app/views/Profile/IndividualProfile/index.tsx [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+    const handleWishlistClick = useCallback(() => {
+        console.warn('handle me');
+    }, []);
+
+    const handleCartClick = useCallback(() => {
+        console.warn('handle me');
+    }, []);
+
+    const orderListRendererParams = useCallback((_, data: OrderListQuery) => ({
```
Comment (AdityaKhatri) [inline app/views/Profile/IndividualProfile/index.tsx]:
```suggestion
    const orderListRendererParams: OrderListRendererProps = useCallback((_, data: OrderListQuery) => ({
```

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]
@subinasr if multiple people are working on the profiles. Lets wrap and merge this one up as soon as possible.

File: app/Base/configs/routes.tsx (hunk 1 above)
Comment (subinasr) [reply app/Base/configs/routes.tsx]: For some reason, when I change the visibility to 'is-authenticated', the URL '/profile/' redirects to the home page. Need to understand this.

File: app/views/OrderList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+interface BookListProps {
+    title: string;
+    quantity: number;
+    isbn: string;
+    price: number;
+    edition: string;
+    image: string;
+}
+
+function BookListRenderer(props: BookListProps) {
```
Comment (AdityaKhatri) [inline app/views/OrderList/index.tsx]: Rename to Book

File: app/views/Profile/index.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    const [userType, setUserType] = useState<UserMeType | undefined>();
+    useQuery<MyProfileQuery, MyProfileQueryVariables>(
+        MY_PROFILE,
+        {
+            onCompleted: (data) => {
+                setUserType(data?.me?.userType);
+            },
+        },
+    );
+
```
Comment (AdityaKhatri) [inline app/views/Profile/index.tsx]: read directly

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/Base/components/Navbar/index.tsx [hunk 6]
```diff
@@ -140,10 +140,14 @@ function Navbar(props: Props) {
                                 Hello
                             </div>
                             <div>
-                                <strong>
-                                    {user.displayName}
-                                </strong>
-                                !
+                                <Link
+                                    to="/profile/"
```
Comment (AdityaKhatri) [inline app/Base/components/Navbar/index.tsx]: use from routes, use SmartLink

File: app/Base/configs/routes.tsx [hunk 7]
```diff
@@ -40,14 +40,13 @@ const home = wrap({
...
+    path: '/profile/',
     title: 'My Profile',
     navbarVisibility: true,
-    component: lazy(() => import('#views/Template')),
+    component: lazy(() => import('#views/Profile')),
     componentProps: {
-        name: 'My Profile Page',
     },
-    visibility: 'is-authenticated',
+    visibility: 'is-authenticated', // TODO: change visibility to 'is-authenticated'
```
Comment (AdityaKhatri) [inline app/Base/configs/routes.tsx]: remove TODO

File: app/views/OrderList/index.tsx [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+    const {
+        activeOrderId,
+    } = props;
+
+    const [pageSize, setPageSize] = useState<number>(MAX_ITEMS_PER_PAGE);
+    const [page, setPage] = useState<number>(1);
+    const orderVariables = useMemo(() => ({
+        pageSize,
+        page,
+    }), []);
```
Comment (AdityaKhatri) [inline app/views/OrderList/index.tsx]: send this here

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**